services/QuillLMS/app/controllers/translated_texts_controller.rb

```diff
@@ -5,12 +5,14 @@ class TranslatedTextsController < ApplicationController
 
   def index
     @translated_texts = TranslatedText
-      .all
       .includes(:english_text, :translation_mappings)
+      .order(created_at: :asc)
       .page(params[:page])
       .per(200)
 
+    @translated_texts = @translated_texts.where(locale: params[:locale]) if params[:locale].present?
```

Comment:
This method is calling to the DB twice, when it only needs one call. One way to fix this, while staying in the AREL syntax, would be:

```
    where_clause = params[:locale].present? ? ['local IN ?', params[:locale]] : ['1 = ?', '1']
    @translated_texts = TranslatedText
      .includes(:english_text, :translation_mappings)
      .where(*where_clause)
      .order(created_at: :asc)
      .page(params[:page])
      .per(200)
```

Reply:
Rails will lazy load this, so it only does one query, even though it looks like two!

Reply:
My bad; my rails console inspect was causing the non-lazy load; doing this fixed the issue:

`console$ <query> ; nil`